iroha/src/sumeragi.rs

```diff
@@ -123,209 +119,122 @@ impl Sumeragi {
-        transactions: &[AcceptedTransaction],
-    ) -> Result<(), String> {
-        log::debug!(
-            "{:?} - Gossiping transactions. Number of transactions to forward: {}",
-            self.network_topology.role(&self.peer_id),
-            transactions.len(),
-        );
-        let leader = self.network_topology.leader().clone();
-        let this_peer = self.peer_id.clone();
-        let peers = self.network_topology.sorted_peers.clone();
-        let transactions = transactions.to_vec();
-        let mut send_futures = Vec::new();
-        // TODO: send transactions in batch not to crowd message channels.
-        for peer in &peers {
-            for transaction in &transactions {
-                if peer != &leader && peer != &this_peer {
-                    send_futures.push(
-                        Message::from(TransactionForwarded::new(transaction, &this_peer))
-                            .send_to(peer),
-                    );
-                }
-            }
-        }
-        let results = futures::future::join_all(send_futures).await;
-        results
-            .into_iter()
-            .filter_map(|result| result.err())
-            .for_each(|error| log::error!("Failed to gossip transactions: {:?}", error));
-        Ok(())
-    }
-
-    /// Should be called by a leader to start the consensus round with `BlockCreated` message.
-    pub async fn validate_and_publish_created_block(
-        &mut self,
-        block: ChainedBlock,
-    ) -> Result<(), String> {
-        let wsv = self.world_state_view.clone();
-        let block = block.validate(&*wsv.read().await, &self.permissions_validator);
-        let network_topology = self.network_topology_current_or_genesis(&block);
-        log::info!(
-            "{:?} - Created a block with hash {}.",
-            network_topology.role(&self.peer_id),
-            block.hash(),
-        );
-        for event in Vec::<Event>::from(&block.clone()) {
-            self.events_sender.send(event).await;
-        }
-        if !network_topology.is_consensus_required() {
-            self.commit_block(block).await;
-            Ok(())
-        } else {
-            *self.voting_block.write().await = Some(VotingBlock::new(block.clone()));
-            let message = Message::BlockCreated(block.clone().sign(&self.key_pair)?.into());
-            let recipient_peers = network_topology.sorted_peers.clone();
-            let this_peer = self.peer_id.clone();
-            let mut send_futures = Vec::new();
-            for peer in &recipient_peers {
-                if this_peer != *peer {
-                    send_futures.push(message.clone().send_to(peer));
-                }
+                });
             }
             let results = futures::future::join_all(send_futures).await;
             results
-                .into_iter()
-                .filter_map(|result| result.err())
+                .iter()
+                .filter(|result| result.is_err())
```

Comment:
Here `filter_map(|result| result.err())` would suit better, as it filters Ok-es and unwraps error